src/main/java/org/spongepowered/api/registry/RegistryEntry.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.registry;
+
+import org.spongepowered.api.ResourceKey;
+
+/**
+ * Represents an entry within a {@link Registry}.
+ *
+ * @param <V> The type of the registry
+ */
+public interface RegistryEntry<V> {
+
+    /**
+     * Gets the {@link ResourceKey key} this entry exists under in a {@link Registry}.
+     *
+     * @return The key
+     */
+    ResourceKey key();
```

Comment:
`RegistryKey`? Could have too, but this isn't sufficient for identification purposes.

Reply:
`RegistryKey` could be done here but if you have an entry, you at some point had/know of the actual `Registry` it came from.

Plugin devs are not expected to ever build entries.